explore/src/main/scala/explore/observationtree/ObsList.scala

```diff
@@ -60,121 +55,112 @@ object ObsList {
       ObsSummaryWithTargetsAndConstraints.id
     )
 
-  protected class Backend {
-    protected def insertObs(
-      pos:        Int,
-      focusedObs: ReuseView[Option[Observation.Id]],
-      undoCtx:    UndoContext[ObservationList]
-    )(implicit
-      c:          TransactionalClient[IO, ObservationDB]
-    ): Callback = {
-      // Temporary measure until we have id pools.
-      val newObs = CallbackTo(Random.nextInt(0xfff)).map(int =>
-        ObsSummaryWithTargetsAndConstraints(
-          Observation.Id(PosLong.unsafeFrom(int.abs.toLong + 1)),
-          targets = List.empty,
-          constraints = ConstraintsSummary.default,
-          ObsStatus.New,
-          ObsActiveStatus.Active,
-          Duration.ZERO
-        )
-      )
+  protected def insertObs(
+    pos:        Int,
+    focusedObs: ReuseView[Option[Observation.Id]],
+    undoCtx:    UndoContext[ObservationList],
+    adding:     ReuseView[Boolean]
+  )(implicit
+    ctx:        AppContextIO
+  ): IO[Unit] =
+    adding.set(true).to[IO] >>
+      createObservation[IO]().flatMap {
+        _.fold(IO.unit) { obs =>
+          ObsListActions
+            .obsExistence(obs.id, focusedObs)
+            .mod(undoCtx)(obsListMod.upsert(obs, pos))
+            .to[IO]
+        }
+      } >> adding.set(false).to[IO]
```

Comment:
We can use `guarantee` here to make sure the UI doesn't get stuck in case there's an error communicating with the ODB.

This is actually a more general concern that we should address at some point, ie recovering from errors and showing error messages.